src/main/kotlin/org/rust/lang/core/psi/impl/RustNumericLiteralImpl.kt

```diff
@@ -1,13 +1,95 @@
 package org.rust.lang.core.psi.impl
 
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElementVisitor
 import com.intellij.psi.tree.IElementType
 import org.rust.lang.core.psi.RustLiteral
 import org.rust.lang.core.psi.RustLiteralTokenType
+import org.rust.lang.core.psi.RustTokenElementTypes.FLOAT_LITERAL
+import org.rust.lang.core.psi.RustTokenElementTypes.INTEGER_LITERAL
+import org.rust.lang.core.psi.visitors.RustVisitorEx
 
 class RustNumericLiteralImpl(type: IElementType, text: CharSequence) : RustLiteral.Number(type, text) {
+    override val valueAsLong: Long?
+        get() = this.valueString
+            ?.filter { it != '_' }
+            ?.let {
+                val (start, radix) = when (it.take(2)) {
+                    "0b" -> 2 to 2
+                    "0o" -> 2 to 8
+                    "0x" -> 2 to 16
+                    else -> 0 to 10
+                }
+                try {
+                    java.lang.Long.parseUnsignedLong(it.substring(start), radix)
```

Comment:
We have not switched to Java 8 yet, and complains from early days that "I can't build this plugin on Java7" suggest that a non negligible fraction of users uses Java 7 to run IDEA. `Long.parseLong` and a `TODO` would be fine.

Also a comment that all literals are unsigned, and that `-92` is an unary expression would be helpful here.